app/scripts/modules/kubernetes/src/pipeline/stages/runJob/runJobStage.js

```diff
@@ -76,47 +89,79 @@ module.exports = angular.module('spinnaker.kubernetes.pipeline.stage.runJobStage
         this.accounts = accounts;
       });
 
-    this.stage.cloudProvider = 'kubernetes';
-    this.stage.application = $scope.application.name;
-
-    if (!this.stage.credentials && $scope.application.defaultCredentials.kubernetes) {
-      this.stage.credentials = $scope.application.defaultCredentials.kubernetes;
-    }
-
-    this.onChange = (changes) => {
-      this.stage.container.imageDescription.registry = changes.registry;
-    };
 
-    this.triggerImages = () => {
+    this.searchImages = (query) => {
+      kubernetesImageReader.findImages({
+        provider: 'dockerRegistry',
+        count: 50,
+        q: query
+      }).then((data) => {
+        
+        if (this.pipeline.triggers) {
+          data = data.concat(this.pipeline.triggers.map((image) => {
+            image.fromTrigger = true;
+            return image;
+          }));
+        }
 
-      if (this.pipeline.triggers.length <= 0) {
-        return [];
-      }
+        if (this.contextImages) {
+          data = data.concat(this.contextImages);
+        }
+        
+        this.containers = _.map(data, (image) => {
+          return {
+            name: image.repository.replace(/_/g, '').replace(/[\/ ]/g, '-').toLowerCase(),
+            imageDescription: {
+              repository: image.repository,
+              tag: image.tag,
+              registry: image.registry,
+              fromContext: image.fromContext,
+              fromTrigger: image.fromTrigger,
+              fromFindImage: image.fromFindImage,
+              cluster: image.cluster,
+              account: image.account,
+              pattern: image.pattern,
+              stageId: image.stageId,
+            },
+            imagePullPolicy: 'IFNOTPRESENT',
+            account: image.accountName,
+            requests: {
+              memory: null,
+              cpu: null,
+            },
+            limits: {
+              memory: null,
+              cpu: null,
+            },
+            ports: [
+            {
```

Comment:
Weird formatting here (probably taken from something I did). I would move the opening and closing brackets and braces together:
```
ports: [{
  ...
}],
```